.github/workflows/nightly-release.yml

```diff
@@ -0,0 +1,101 @@
+name: Nightly Release
+
+on:
+  # TODO: Remove once this workflow is in the main branch
+  push:
+    branches: [ add_nightly_release ]
+  workflow_dispatch:
+
+jobs:
+  rtl-repo-sync:
+    name: RTL Repo Sync
+    uses: ./.github/workflows/rtl-repo-sync.yml
+
+  find-latest-release:
+    name: Find Latest Release
+    needs: rtl-repo-sync
+    runs-on: ubuntu-22.04
+    outputs:
+      create_release: ${{ steps.find.outputs.create_release }}
+      new_release_tag: ${{ steps.find.outputs.new_release_tag }}
+      release_ref: ${{ steps.find.outputs.release_ref }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+          ref: 'main'
+          fetch-depth: 0
+
+      - name: Find latest release
+        id: find
+        run: |
+          DATE="$(date +'%Y%m%d')"
+          TAG_PREFIX="release_v"
+          TAG_BASE="${TAG_PREFIX}${DATE}_"
+          INDEX=0
+          while git tag | grep ${TAG_BASE}${INDEX}; do
+              ((INDEX+=1))
+          done
+          MOST_RECENT_RELEASE=None
+          if git tag | grep ${TAG_PREFIX} > /dev/null; then
+              MOST_RECENT_RELEASE=$(git tag | grep ${TAG_PREFIX} | sort -r | head -1)
+          fi
+          if [[ "$MOST_RECENT_RELEASE" == "None" ]]; then
+              echo "create_release=true" >> $GITHUB_OUTPUT
+          else
+              COMMITS_AFTER_LAST_RELEASE=$(git rev-list --count $MOST_RECENT_RELEASE..HEAD)
+              if [[ $COMMITS_AFTER_LAST_RELEASE -gt 0 ]]; then
+                  echo "create_release=true" >> $GITHUB_OUTPUT
+              else
+                  echo "create_release=false" >> $GITHUB_OUTPUT
+              fi
+          fi
+          echo "new_release_tag=${TAG_BASE}${INDEX}" >> $GITHUB_OUTPUT
+          echo "release_ref=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+          echo "Current ref $(git rev-parse HEAD) will receive tag ${TAG_BASE}${INDEX} after tests"
+
+  verilator-full-suite:
+    name: Temp
+    needs: find-latest-release
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Get current date
+        run: echo "date=$(date +'%Y%m%d')" >> $GITHUB_OUTPUT
+
+  #verilator-full-suite:
```

Comment:
Note: Uncomment this and remove the temporary verilator-full-suite above before merging!